prow/config/config_test.go

```diff
@@ -4165,3 +4165,102 @@ func TestDefaultAndValidateReportTemplate(t *testing.T) {
 		})
 	}
 }
+
+func TestValidatePresubmits(t *testing.T) {
+	t.Parallel()
+	testCases := []struct {
+		name          string
+		presubmits    []Presubmit
+		expectedError string
+	}{
+		{
+			name: "Duplicate context causes error",
+			presubmits: []Presubmit{
+				{JobBase: JobBase{Name: "a"}, Reporter: Reporter{Context: "repeated"}},
+				{JobBase: JobBase{Name: "b"}, Reporter: Reporter{Context: "repeated"}},
+			},
+			expectedError: `[jobs b and a report to the same GitHub context "repeated", jobs a and b report to the same GitHub context "repeated"]`,
+		},
+		{
+			name: "Duplicate jobname causes error",
+			presubmits: []Presubmit{
+				{JobBase: JobBase{Name: "a"}, Reporter: Reporter{Context: "foo"}},
+				{JobBase: JobBase{Name: "a"}, Reporter: Reporter{Context: "bar"}},
+			},
+			expectedError: "duplicated presubmit job: a",
```

Comment:
We technically allow this if they don't intersect on branch config, https://github.com/kubernetes/test-infra/issues/9948

Reply:
Yeah I was lazy. Added a test for that as well.